core/kernel/persistence/smoothsql/class.Resource.php

```diff
@@ -838,22 +843,41 @@ private function buildTrippleArray(
     private function normalizePropertyValues($value)
     {
         $normalizedValues = [];
-        if ($value instanceof core_kernel_classes_Resource) {
-            $normalizedValues[] = $value->getUri();
-        } elseif (is_array($value)) {
+
+        if (is_array($value)) {
             foreach ($value as $val) {
-                if ($val !== null) {
-                    $normalizedValues[] = $val instanceof core_kernel_classes_Resource
-                        ? $val->getUri()
-                        : $val;
+                if ($val === null) {
+                    continue;
                 }
+                $normalizedValues[] = $this->normalizeSinglePropertyValue($val);
             }
         } else {
-            $normalizedValues[] = ($value == null) ? '' : $value;
+            $normalizedValues[] = $this->normalizeSinglePropertyValue($value);
         }
+
         return $normalizedValues;
     }
 
+    private function normalizeSinglePropertyValue($value)
+    {
+        if ($value === null) {
+            return '';
+        }
+
+        if ($value instanceof core_kernel_classes_Resource) {
+            return $value->getUri();
```

Comment:
you just return the value, shouldn't it also be checked & normalized?

Reply:
URI is identification which is generated on the TAO side, not needed to normalized